### ISSUE TITLE
helm v3a2 helm template --replace fails with cannot re-use a name that is still in use

### ISSUE THREAD

**Issue by nwsparks** [helm/helm#6101]
Output of `helm version`:

`version.BuildInfo{Version:"v3.0.0-alpha.2", GitCommit:"97e7461e41455e58d89b4d7d192fed5352001d44", GitTreeState:"clean", GoVersion:"go1.12.7"}`

Output of `kubectl version`:
```
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.0", GitCommit:"e8462b5b5dc2584fdcd18e6bcfe9f1e4d970a529", GitTreeState:"clean", BuildDate:"2019-06-19T16:40:16Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}

Server Version: version.Info{Major:"1", Minor:"13+", GitVersion:"v1.13.7-eks-c57ff8", GitCommit:"c57ff8e35590932c652433fab07988da79265d5b", GitTreeState:"clean", BuildDate:"2019-06-07T20:43:03Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.):  `eks`

Attempts to render templates for a helm chart that is already installed fail with `Error: cannot re-use a name that is still in use` even when `--replace` is passed.

Output:

```
$ helm template nginx-ingress stable/nginx-ingress --values nginx-ingress.yml --replace  --debug
printer.go:84: [debug] Original chart version: ""
printer.go:84: [debug] CHART PATH: /home/user/.helm/cache/archive/nginx-ingress-1.8.1.tgz

Error: cannot re-use a name that is still in use
helm.go:76: [debug] cannot re-use a name that is still in use
helm.sh/helm/pkg/action.(*Install).availableName
        /go/src/helm.sh/helm/pkg/action/install.go:283
helm.sh/helm/pkg/action.(*Install).Run
        /go/src/helm.sh/helm/pkg/action/install.go:122
main.runInstall
        /go/src/helm.sh/helm/cmd/helm/install.go:213
main.newTemplateCmd.func1
        /go/src/helm.sh/helm/cmd/helm/template.go:53
helm.sh/helm/vendor/github.com/spf13/cobra.(*Command).execute
        /go/src/helm.sh/helm/vendor/github.com/spf13/cobra/command.go:762
helm.sh/helm/vendor/github.com/spf13/cobra.(*Command).ExecuteC
        /go/src/helm.sh/helm/vendor/github.com/spf13/cobra/command.go:850
helm.sh/helm/vendor/github.com/spf13/cobra.(*Command).Execute
        /go/src/helm.sh/helm/vendor/github.com/spf13/cobra/command.go:800
main.main
        /go/src/helm.sh/helm/cmd/helm/helm.go:75
runtime.main
        /usr/local/go/src/runtime/proc.go:200
runtime.goexit
        /usr/local/go/src/runtime/asm_amd64.s:1337
```

**Comment by ian-howell** [comment]
This problem extends to `helm install` as well.

**Comment by bacongobbler** [comment]
fixed in #6122
